raytracing/rays.py

```diff
@@ -99,46 +99,40 @@ def display(self, title="Intensity profile", showTheta=True):
         axis1 = axes[0]
         axis2 = axes[1]
 
-        (x,y) = self.rayCountHistogram()
-        #axis1.set_title('Intensity profile')
-        axis1.plot(x,y,'k-',label="Intensity")
-        axis1.set_ylim([0, max(y)*1.1])
+        (x, y) = self.rayCountHistogram()
+        # axis1.set_title('Intensity profile')
+        axis1.plot(x, y, 'k-', label="Intensity")
+        axis1.set_ylim([0, max(y) * 1.1])
         axis1.set_xlabel("Distance")
         axis1.set_ylabel("Ray count")
         axis1.legend(["Intensity"])
 
-        (x,y) = self.rayAnglesHistogram()
-        #axis2.set_title('Angle histogram')
-        axis2.plot(x,y,'k--',label="Orientation profile")
-        axis2.set_ylim([0, max(y)*1.1])
-        axis2.set_xlim([-pi/2,pi/2])
+        (x, y) = self.rayAnglesHistogram()
+        # axis2.set_title('Angle histogram')
+        axis2.plot(x, y, 'k--', label="Orientation profile")
+        axis2.set_ylim([0, max(y) * 1.1])
+        axis2.set_xlim([-pi / 2, pi / 2])
         axis2.set_xlabel("Angles [rad]")
         axis2.set_ylabel("Ray count")
         axis2.legend(["Angle"])
 
         plt.show()
-    
+
     def displayProgress(self):
         nRays = len(self)
         if self.iteration % self.progressLog == 0:
             self.progressLog *= 3
             if self.progressLog > nRays:
                 self.progressLog = nRays
 
-            print("Progress {0}/{1} ({2:.0f}%) ".format(self.iteration, nRays, self.iteration/nRays*100))
-
-    def __len__(self) -> int:
```

Comment:
I understand that this looks unnecessary : 
We want `Rays` to be iterable, and we only need to overload `__iter__` and `__next__`: https://docs.python.org/3/library/stdtypes.html#typeiter

However, `Rays` is the base class for `RandomRays`, which generate their rays on-demand up to a maximum `maxCount` and don't have an array with elements in it when created.  My goal was to have both Ray and Rays appear similar, which is why I added __len__ to Rays, even if it looks redundant: we can always do `len(Rays)` regardless of the class or subclass.

Reply:
Ok I see, I will make the changes